docroot/modules/humsci/hs_capx/src/Capx.php

```diff
@@ -0,0 +1,262 @@
+      $url .= '&includeChildren=true';
+    }
+    return $url;
+  }
+
+  /**
+   * Get the url for CAPx for given workgroups.
+   *
+   * @param string $workgroups
+   *   Commas separated list of workgroups.
+   *
+   * @return string
+   *   CAPx URL.
+   */
+  public static function getWorkgroupUrl($workgroups) {
+    $workgroups = preg_replace('/[^A-Z,:]/', '', strtoupper($workgroups));
+    return self::CAP_URL . "?privGroups=$workgroups&ps=1000";
+  }
+
+  /**
+   * Test the connection with the username and passwords is valid.
+   *
+   * @return bool
+   *   The connection was successful.
+   */
+  public function testConnection() {
+    $parameters = ['grant_type' => 'client_credentials'];
+
+    try {
+      $response = $this->client->request('GET', self::AUTH_URL, [
+        'query' => $parameters,
+        'auth' => [$this->username, $this->password],
+      ]);
+    }
+    catch (GuzzleException $e) {
+      // Most errors originate from the API itself.
+      $this->logger->error($e->getMessage());
+      return FALSE;
+    }
+
+    // Just in case the API doesn't respond with an error, we also check the
+    // status code.
+    return $response->getStatusCode() == 200;
+  }
+
+  /**
+   * Sync the organization database with the api data from CAP.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  public function syncOrganizations() {
+    $this->insertOrgData($this->getOrgData());
+  }
+
+  /**
+   * Insert the given organization data into the database.
+   *
+   * @param array $org_data
+   *   Keyed array of organization data.
+   * @param array $parent
+   *   The organization parent if one exists.
+   *
+   * @throws \Exception
+   */
+  protected function insertOrgData(array $org_data, array $parent = []) {
+    if (!empty($org_data['children'])) {
+      foreach ($org_data['children'] as $child) {
+        $this->insertOrgData($child, $org_data);
+      }
+    }
+
+    $insert_data = [
+      'name' => $org_data['name'],
+      'alias' => $org_data['alias'],
+      'orgcodes' => serialize($org_data['orgCodes']),
+      'parent' => $parent ? $parent['alias'] : '',
+    ];
+
+    $this->database->merge('hs_capx_organizations')
+      ->fields($insert_data)
+      ->key('alias', $insert_data['alias'])
+      ->execute();
+  }
+
+  /**
+   * Get the organization data array from the API.
+   *
+   * @return array
+   *   Keyed array of all organization data.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   */
+  protected function getOrgData() {
+    if ($cache = $this->cache->get('capx:org_data')) {
+      return $cache->data;
+    }
+
+    $token = $this->getAccessToken();
+
+    // AA00 is the root level of all Stanford.
+    $result = $this->client->request('GET', self::API_URL . '/cap/v1/orgs/AA00', ['query' => ['access_token' => $token]]);
+    $result = json_decode($result->getBody()->getContents(), TRUE);
+    $this->cache->set('capx:org_data', $result, time() + 60 * 80 * 24 * 7, [
```

Comment:
60 * `80` * 24 * 7 ?

Reply:
:man_facepalming: it gives extra 2.3 days of cache haha.